tests/call_tests.cpp

```diff
@@ -0,0 +1,295 @@
+  BOOST_REQUIRE_EXCEPTION(call_test(token_addr, 1234, "alice"_n, "alice"_n),
+                          eosio_assert_message_exception, eosio_assert_message_is("decrementing more than available"));
+
+  // Transfer enough funds
+  transfer_token("alice"_n, "evm"_n, make_asset(1000000), "alice");
+
+  BOOST_REQUIRE(intx::uint256(vault_balance("alice"_n)) == intx::exp<256>(10, 18) * 100);
+  auto total_fund2 = intx::uint256(vault_balance("alice"_n)) + intx::uint256(vault_balance("evm"_n));
+
+  BOOST_REQUIRE_EXCEPTION(call_test(token_addr, 0, "alice"_n, "alice"_n),
+                          eosio_assert_message_exception, eosio_assert_message_is("inline actions must succeed"));
+
+
+  // Call and check results 
+  call_test(token_addr, 1234, "alice"_n, "alice"_n);
+  auto count = get_count(token_addr);
+  BOOST_REQUIRE(count == 1234);
+
+  // Gas go from alice's vault to evm's vault
+  BOOST_REQUIRE(total_fund2 == intx::uint256(vault_balance("alice"_n)) + intx::uint256(vault_balance("evm"_n)));
+
+
+  // Advance block so we do not generate same transaction.
+  produce_block();
+
+  call_test(token_addr, 4321, "alice"_n, "alice"_n);
+  count = get_count(token_addr);
+  BOOST_REQUIRE(count == 5555);
+
+  // Gas go from alice's vault to evm's vault
+  BOOST_REQUIRE(total_fund2 == intx::uint256(vault_balance("alice"_n)) + intx::uint256(vault_balance("evm"_n)));
+
+  // Function being called on behalf of reserved address of eos account "alice"
+  auto caller = get_lastcaller(token_addr);
+  BOOST_REQUIRE(caller == make_reserved_address("alice"_n.to_uint64_t()));
+
+} FC_LOG_AND_RETHROW()
+
+BOOST_FIXTURE_TEST_CASE(admincall_test_function, call_evm_tester) try {
+  evm_eoa evm1;
+  evm_eoa evm2;
+
+  auto total_fund = intx::uint256(vault_balance("evm"_n));
+
+  // Fund evm1 address with 100 EOS
+  transfer_token("alice"_n, "evm"_n, make_asset(1000000), evm1.address_0x());
+
+  auto evm1_balance = evm_balance(evm1);
+  BOOST_REQUIRE(!!evm1_balance);
+  BOOST_REQUIRE(*evm1_balance == intx::exp<256>(10, 18) * 100);
+  total_fund += *evm1_balance;
+
+  // Deploy contract
+  auto token_addr = deploy_test_contract(evm1);
+
+  // Deployment gas fee go to evm vault
+  BOOST_REQUIRE(*evm_balance(evm1) + intx::uint256(vault_balance("evm"_n)) == total_fund);
+
+  // Missing authority
+  BOOST_REQUIRE_EXCEPTION(admincall_test(token_addr, 1234, evm2, "alice"_n),
+                          missing_auth_exception, eosio::testing::fc_exception_message_starts_with("missing authority"));
+
+  // Account not created
+  BOOST_REQUIRE_EXCEPTION( admincall_test(token_addr, 1234, evm2, "evm"_n),
+                          eosio_assert_message_exception, eosio_assert_message_is("invalid address"));
+
+  // Transfer small amount to create account
+  transfer_token("alice"_n, "evm"_n, make_asset(100), evm2.address_0x());
+
+  auto evm2_balance = evm_balance(evm2);
+  BOOST_REQUIRE(!!evm2_balance);
+  BOOST_REQUIRE(*evm2_balance == intx::exp<256>(10, 18 - 4) * 100);
```

Comment:
Alternative option: `100 * 100_szabo` where `100_szabo` represents the smallest unit of EOS: 0.0001 EOS.